modules/phase_field/include/postprocessors/FeatureFloodCount.h

```diff
@@ -65,6 +65,9 @@ class FeatureFloodCount : public GeneralPostprocessor,
   /// Returns a Boolean indicating whether this feature intersects _any_ boundary
   virtual bool doesFeatureIntersectBoundary(unsigned int feature_id) const;
 
+  /// Returns a Boolean indicating whether this feature intersects boundaries in a user-supplied list
+  virtual bool doesFeatureIntersectSpecifiedBoundary(unsigned int feature_id) const;
```

Comment:
It really feel like the API should allow the user to supply the list of boundaries as a parameter. However that would fundamentally change the usage in the input file too. Thoughts?

I guess this is fine if you like it how it is.

Reply:
I have it set up so that the user specifies a list of boundaries as the optional parameter `specified_boundaries` in FeatureFloodCount, so it seems like it's allowing what you suggest- maybe I am misunderstanding what you are asking though. 

Is the function name `doesFeatureIntersectSpecifiedBoundary` possibly misleading in that it implies only a single boundary can be considered?

Reply:
What I was suggesting was an API that looked like this because it makes sense (just based off the name):
```
virtual bool doesFeatureIntersectSpecifiedBoundary(unsigned int feature_id, std::vector<BoundaryID> boundaries) const;
```
You'd actually be specifying which boundaries you were interested in through the method call. The problem with this, is that it means that the source of those boundaries would be coming in from a different object. In this case the VectorPostprocessor, or an AuxKernel that would be consuming that information. That's a big fundamental change because you'd have to put the boundaries of interest in a different block in your input file. The up side to that would be that you could potentially have a whole series of different vector postprocessors each grabbing information for a specific boundary. So that API would actually be better from a design point of view. 

This design means that the specified boundaries are bound to the instance of the FeatureFloodCounter or GrainTracker objects and can't be changed. The API just looks weird. I'm OK with this, just talking the design through.

Reply:
OK I see. For the time being I don't have a use case that would necessitate multiple VPP's as you mention, so I think I'm OK with the current implementation. But we can keep that in mind for future changes if the need comes up.